### PR TITLE
[NATS] Add routes data stream dimensions and metric types

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds `routes` data stream dimensions and metric types.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by gizas** [comment]
Same in all relevant PRs you had opened: Can you please provide some screenshots or eveidence of tests for your changes?

**Comment by gpop63** [comment]
After fixing aggregation type for counters. @gizas 

scrnli_9_21_2023_3-01-07 PM

**Comment by gpop63** [comment]
<details>
<summary>Rollover test</summary>

```json
{
  "acknowledged": true,
  "shards_acknowledged": true,
  "old_index": ".ds-metrics-nats.routes-default-2023.09.21-000001",
  "new_index": ".ds-metrics-nats.routes-default-2023.09.21-000002",
  "rolled_over": true,
  "dry_run": false,
  "conditions": {}
}
```

</details> 

<details>
<summary>Mappings</summary>

```json
"nats": {
          "properties": {
            "routes": {
              "properties": {
                "total": {
                  "type": "long",
                  "time_series_metric": "gauge"
                }
              }
            },
```

</details>

File: packages/nats/data_stream/routes/fields/fields.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   release: ga
   fields:
     - name: total
+      metric_type: gauge
```
Comment (agithomas) [inline packages/nats/data_stream/routes/fields/fields.yml:6]: Can you please validate if it is counter type?

File: packages/nats/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.3.7"
+  changes:
+    - description: Added dimensions and metric_type mappings to routes data stream.
```
Comment (agithomas) [inline packages/nats/changelog.yml:4]:
```suggestion
    - description: Add dimensions and metric_type mappings to routes data stream.
```

File: packages/nats/data_stream/routes/fields/fields.yml (hunk 1 above)
Comment (gpop63) [reply packages/nats/data_stream/routes/fields/fields.yml:6]: I have 2 routes and I disconnected 1 route.

Before:
```json
    "nats": {
      "server": {
        "id": "YbOxX5huJdagJNMQHh1fE3",
        "time": "2023-10-12T13:27:37.489166157Z"
      },
      "routes": {
        "total": 2
      }
    },
```

After disconnecting route:
```json
    "nats": {
      "routes": {
        "total": 1
      },
      "server": {
        "id": "YbOxX5huJdagJNMQHh1fE3",
        "time": "2023-10-12T13:27:47.49019478Z"
      }
    },
```

Adding the route again:
```json
    "nats": {
      "routes": {
        "total": 2
      },
      "server": {
        "id": "YbOxX5huJdagJNMQHh1fE3",
        "time": "2023-10-12T13:30:37.489245742Z"
      }
    },
```

**Review by constanca-m — APPROVED** [review]

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**